validator.go

```diff
@@ -82,29 +82,28 @@ func (t *TableValidator) validateRow(row []string) error {
 // a set of validators for each field.
 func (t *TableValidator) Init() error {
 	var (
-		err  error
-		head []string
+		err         error
+		head        []string
+		length_err  = false
+		match_err   = false
```

Comment:
Also variables that are declared with the [zero value](https://golang.org/ref/spec#The_zero_value) can be just declared, `var matchErr bool` without the assignment.